paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt

```diff
@@ -18,26 +18,36 @@ package app.cash.paparazzi.gradle
 import com.android.build.gradle.BaseExtension
 import com.android.ide.common.symbols.getPackageNameFromManifest
 import org.gradle.api.DefaultTask
-import org.gradle.api.Project
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
 import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.PathSensitive
 import org.gradle.api.tasks.PathSensitivity
 import org.gradle.api.tasks.TaskAction
-import java.io.File
 
 @CacheableTask
 open class PrepareResourcesTask : DefaultTask() {
+  @get:Input
+  internal lateinit var packageName: Property<String>
+  @get:Input
+  internal lateinit var targetSdkVersion: Property<String>
+  @get:Input
+  internal lateinit var compileSdkVersion: Property<String>
+
   @get:InputDirectory
   @get:PathSensitive(PathSensitivity.RELATIVE)
   internal val mergeResourcesOutput: DirectoryProperty = project.objects.directoryProperty()
+  private val mergeResourcesOutputPath =
+    project.relativePath(mergeResourcesOutput.get().asFile.path)
 
   @get:InputDirectory
   @get:PathSensitive(PathSensitivity.RELATIVE)
   internal val mergeAssetsOutput: DirectoryProperty = project.objects.directoryProperty()
+  private val mergeAssetsOutputPath = project.relativePath(mergeAssetsOutput.get().asFile.path)
```

Comment:
I think we can revert both and keep them inlined, but replace:

```
project.relativePath(dirProperty.get().asFile.path)
```

with

```
private val projectDirectory = project.layout.projectDirectory
```
outside of the TaskAction, and
```
projectDirectory.file(dirProperty.get().asFile.path).asFile.path
```
inside?

Reply:
actually that won't work, because it'll lose the relativism of the paths, but this works!


```
  internal val mergeAssetsOutput: DirectoryProperty = project.objects.directoryProperty()

  private val projectDirectory = project.layout.projectDirectory

        ...
        it.write(projectDirectory.relativize(mergeAssetsOutput.get()))
        ...

  fun Directory.relativize(child: Directory) = asFile.toPath().relativize(child.asFile.toPath()).toString()
```

Reply:
15b6d42